### PR TITLE
Move roughness limiter and sort into their own classes

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This PR moves the roughness limiter and sort effects into their own classes, retiring the effects class.

### CODE REVIEW

**Comment by BastiaanOlij** [comment]
Note, fine to merge after stable release in one of the patch releases.

**Comment by clayjohn** [comment]
Is the roughness limiter code even hooked up to anything right now? I thought that Reduz stopped using it when we added the screenspace roughness code into the main render pass.

That being said, if its not working right now, we should get it working again.

**Comment by BastiaanOlij** [comment]
@clayjohn no it's not used anywhere, I didn't want to delete the code in case we still have plans for it

**Review by clayjohn — COMMENTED** [review]
Seems fine to me. I would update the commented note to say that this should be updated in the future or else someone will come by and delete the entire class for being unused

File: servers/rendering/renderer_rd/effects/roughness_limiter.h [hunk 1]
```diff
@@ -28,106 +28,40 @@
...
 #include "servers/rendering/renderer_scene_render.h"
 
 #include "servers/rendering_server.h"
 
-class EffectsRD {
-private:
-	bool prefer_raster_effects;
+namespace RendererRD {
+
+// Note, this logic seems to be unused, not sure if this is an oversight or something that we plan to implement in the future.
```
[the remark this answers is not in the document]
Comment (BastiaanOlij) [reply servers/rendering/renderer_rd/effects/roughness_limiter.h]: Done :)

**Review by clayjohn — APPROVED** [review]
Looks good to me

**Comment by akien-mga** [comment]
Thanks!

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 2]
```diff
@@ -364,7 +362,7 @@ void SkyRD::ReflectionData::update_reflection_data(int p_size, int p_mipmaps, bo
 			mm.size.height = mmh;
 			mm.view = RD::get_singleton()->texture_create_shared_from_slice(RD::TextureView(), downsampled_radiance_cubemap, 0, j, 1, RD::TEXTURE_SLICE_CUBEMAP);
 			RD::get_singleton()->set_resource_name(mm.view, "Downsampled Radiance Cubemap Mip " + itos(j) + " ");
-			if (prefer_raster_effects) {
+			if (render_buffers_can_be_storage) {
```
Comment (BastiaanOlij) [inline servers/rendering/renderer_rd/environment/sky.cpp:365]: This is a bug we introduced here, fixed in #75664

**Pull request merged**